### ISSUE TITLE
Missing Implementation for Calculating Likelihood Values as per Equation (10)

### ISSUE THREAD

**Issue by Monohydroxides** [louaaron/Score-Entropy-Discrete-Diffusion#2]
Hello, thank you for your outstanding work! In the code, I don't seem to find the implementation for calculating likelihood values as given by equation (10) in the paper. Could you please inform me about when you plan to upload it to the GitHub repository? Or perhaps I missed a part of the code that covers this calculation. Could you please assist me in identifying where the code for calculating likelihood values is located?

**Comment by alexanderswerdlow** [comment]
Following up on this, it seems the only perplexity metric in the repo is using the SEDD predictions as GT and measuring the perplexity of a pre-trained GPT-2 model, presumably to measure alignment with that model.

Would it be at all possible to provide the perplexity metric used in the paper?

Thanks!

**Comment by dariorancati** [comment]
I also would like to know how the perplexity could be computed for a general dataset as claimed in the paper, as opposed to the one implemented in the code which only fairly benchmarks against GPT2. 

This question has now been idle for a while: is there any update on the matter?
